sig-robotics/proposal/Implement-VSLAM-algorithm-based-on-KubeEdge/README.md

```diff
@@ -0,0 +1,85 @@
+- The edge applications will be required to keep track for a  moment when the network is disconnected.
+- The cloud applications will be required to perform loop closing and global mapping tasks.
+
+## 2 Proposal
+
+We propose using one of the classic SLAM frameworks: ORB-SLAM2 as the basis. Just like ORB-SLAM2 or other classic frameworks, the SLAM system mainly includes four modules: tracking, local optimization, loop closure detection and global optimization. 
+
+### 2.1 ORB-SLAM2 Overview
+
+We use ORB-SLAM2 as a prototype since it is open source. So it is necessary to describe the framework of  ORB-SLAM2.
+
+![vslam_kubeedge_orbslam2](images/vslam_kubeedge_orbslam2.png)
+
+ORB-SLAM2 consists of three threads, one per module: tracking, local mapping, and loop closure as shown in above figure. The tracking thread loops through incoming image frames for their initial pose estimation and decides which frames to accept as a key-frame, based on five conditions:
+
+1. If re-localization occurred, then 20 frames should pass to insert a new key-frame.
+2. Either 20 frames have passed after the last inserted key-frame, or local mapping thread is not busy.
+3. The current frame is tracking at least 50 features.
+4. The current frame is tracking fewer than 90% points compared to the frame's reference key-frame
+5. If the current frame tracks less than 100 close points, and can create more than 70 new close points.
+
+### 2.2 Two Strategies
+
+Typically, for real-time localization problem in an **unknown environment**, we have two strategies to achieve cloud-edge collaborative SLAM. they are:
+
+* A complete **open-loop visual odometry system is deployed at the edge**, which generally includes two modules: tracking and local mapping. At the same time, loop closing and global optimization are deployed in the cloud to build a global map.
+* Only the **tracking module is deployed at the edge**, the back-end optimization and loop closing tasks are completed in the cloud, and the cloud continuously pushes local maps to the edge.
+
+The first strategy ensures that the edge has a long-term offline autonomy capability, but increases the computing and memory resource occupation of the edge. On the other hand, how to synchronize local map in the edge and global map in the cloud becomes a new problem.
+
+The second method offloads all optimizations to the cloud, which greatly reduces the pressure on edge resources, but requires strong network stability. At the same time, the edge side only needs to maintain a small-scale local map (below 10 frames), and the local map can be updated easily by clearing local map on the edge and repalcing it with new received local map update from the cloud.
+
+Based on the above discussion, based on the reliable network communication provided by kubeedge and edgemesh and some open source work available for reference(such as [edge-slam][https://github.com/droneslab/edgeslam]), we intend to use the second method for this work.
```

Comment:
Replace: `[edge-slam][https://github.com/droneslab/edgeslam]` to `[edge-slam](https://github.com/droneslab/edgeslam)`